backend/app/feedback/routers.py

```diff
@@ -1,58 +1,61 @@
-from typing import List
-from uuid import UUID  # Import UUID here
+# feedback/routers.py
+from uuid import UUID
 
 from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
 
 from ..auth.dependencies import authenticate_key
-from .schemas import Feedback, FeedbackResponse
+from ..database import get_async_session
+from ..services.FeedbackService import FeedbackService
+from .schemas import FeedbackRequest, FeedbackResponse, ListFeedbackResponse
 
 router = APIRouter(
     dependencies=[Depends(authenticate_key)], tags=["Feedback endpoints"]
 )
 
-# In-memory storage for feedback (can be replaced with a database in production)
-feedback_storage: List[Feedback] = []
-
 
 @router.post("/feedback", response_model=FeedbackResponse)
-async def submit_feedback(feedback: Feedback) -> FeedbackResponse:
+async def submit_feedback(
+    feedback_request: FeedbackRequest,
+    session: AsyncSession = Depends(get_async_session),
+) -> FeedbackResponse:
     """
     This endpoint allows users to submit feedback
+    and saves it using the service layer.
+     and saves it using the service layer.
     """
-
-    feedback_storage.append(feedback)  # Save feedback in the in-memory storage
-    return FeedbackResponse(feedback=feedback)
+    try:
+        saved_feedback = await FeedbackService.submit_feedback(
+            feedback_request, session
+        )
+        return FeedbackResponse(feedback=saved_feedback)
+    except Exception as e:
```

Comment:
Can i suggest capturing the subclass or specific exception here? Usually not best practice to capture the base exception.